src/ParallelAlgorithms/ApparentHorizonFinder/Criteria/Criterion.hpp

```diff
@@ -52,6 +52,9 @@ class Criterion : public PUP::able {
 
   virtual std::string observation_name() = 0;
 
+  virtual bool operator==(const Criterion& other) const = 0;
+  bool operator!=(const Criterion& other) const { return not(*this == other); }
```

Comment:
We've been using `is_equal` or `is_equal_to` in other places. I forget the reason, but in part it's because base classes can't be copied, so it's not clear what it means for them to be equivalent, and `operator==` is "supposed to" be that if you copy `a` then `copy_a == a` is true. Would you mind renaming? It looks like use is ~50/50 in the code between those names, I don't have a preference